admin/projects.go

```diff
@@ -124,6 +126,16 @@ func (s *Service) TeardownProject(ctx context.Context, p *database.Project) erro
 // UpdateProject updates a project and any impacted deployments.
 // It runs a reconcile if deployment parameters (like branch or variables) have been changed and reconcileDeployment is set.
 func (s *Service) UpdateProject(ctx context.Context, proj *database.Project, opts *database.UpdateProjectOptions, reconcileDeployment bool) (*database.Project, error) {
+	if proj.Region != opts.Region || proj.ProdSlots != opts.ProdSlots { // require new deployments
```

Comment:
This adds a redundancy where it will recreate the deployment and then afterwards update it in `impactsDeployments`. The call to `updateDeployment` may even failed because the new deployment is still reconciling (it crudely disregards concurrent reconciles because they're not supported under the hood).

The low down is that any change happening in `UpdateProject` must at most trigger one reconcile on a deployment.

Reply:
We are returning on creating new deployment here so its not triggering another reconcile. Although I think this leads to a bug where other filed aren't updated. I will fix it to read fields from opts.